In [5]:
import pandas as pd
import numpy as np

In [6]:
pf = pd.read_excel('new_performance.xlsx', header=1 )
pf

방송일시      노출(분)    마더코드    상품코드  \
0     2019-01-01 06:00:00  20.000000  100346  201072   
1     2019-01-01 06:00:00        NaN  100346  201079   
2     2019-01-01 06:20:00  20.000000  100346  201072   
3     2019-01-01 06:20:00        NaN  100346  201079   
4     2019-01-01 06:40:00  20.000000  100346  201072   
...                   ...        ...     ...     ...   
38304 2020-01-01 00:20:00  20.000000  100073  200196   
38305 2020-01-01 00:40:00  20.000000  100073  200196   
38306 2020-01-01 01:00:00  20.000000  100073  200196   
38307 2020-01-01 01:20:00  20.000000  100490  201478   
38308 2020-01-01 01:40:00  17.166667  100490  201478   

                            상품명 상품군   판매단가        취급액  
0                 테이트 남성 셀린니트3종  의류  39900  2099000.0  
1                 테이트 여성 셀린니트3종  의류  39900  4371000.0  
2                 테이트 남성 셀린니트3종  의류  39900  3262000.0  
3                 테이트 여성 셀린니트3종  의류  39900  6955000.0  
4                 테이트 남성 셀린니트3종  의류  39900  6672000.0  
...                         ...  ..    ...        ...  
38304  삼성화재 행복한파트너 주택화재보험(1912)  무형      0        NaN  
38305  삼성화재 행복한파트너 주택화재보험(1912)  무형      0        NaN  
38306  삼성화재 행복한파트너 주택화재보험(1912)  무형      0        NaN  
38307        더케이 예다함 상조서비스(티포트)  무형      0        NaN  
38308        더케이 예다함 상조서비스(티포트)  무형      0        NaN  

[38309 rows x 8 columns]

In [7]:
pf['year'] = pf['방송일시'].map(lambda x: x.year)
pf['month'] = pf['방송일시'].map(lambda x: x.month)
pf['day'] = pf['방송일시'].map(lambda x: x.day)
pf['time'] = pf['방송일시'].map(lambda x: x.time())

pf

방송일시      노출(분)    마더코드    상품코드  \
0     2019-01-01 06:00:00  20.000000  100346  201072   
1     2019-01-01 06:00:00        NaN  100346  201079   
2     2019-01-01 06:20:00  20.000000  100346  201072   
3     2019-01-01 06:20:00        NaN  100346  201079   
4     2019-01-01 06:40:00  20.000000  100346  201072   
...                   ...        ...     ...     ...   
38304 2020-01-01 00:20:00  20.000000  100073  200196   
38305 2020-01-01 00:40:00  20.000000  100073  200196   
38306 2020-01-01 01:00:00  20.000000  100073  200196   
38307 2020-01-01 01:20:00  20.000000  100490  201478   
38308 2020-01-01 01:40:00  17.166667  100490  201478   

                            상품명 상품군   판매단가        취급액  year  month  day  \
0                 테이트 남성 셀린니트3종  의류  39900  2099000.0  2019      1    1   
1                 테이트 여성 셀린니트3종  의류  39900  4371000.0  2019      1    1   
2                 테이트 남성 셀린니트3종  의류  39900  3262000.0  2019      1    1   
3                 테이트 여성 셀린니트3종  의류  39900  6955000.0  2019      1    1   
4                 테이트 남성 셀린니트3종  의류  39900  6672000.0  2019      1    1   
...                         ...  ..    ...        ...   ...    ...  ...   
38304  삼성화재 행복한파트너 주택화재보험(1912)  무형      0        NaN  2020      1    1   
38305  삼성화재 행복한파트너 주택화재보험(1912)  무형      0        NaN  2020      1    1   
38306  삼성화재 행복한파트너 주택화재보험(1912)  무형      0        NaN  2020      1    1   
38307        더케이 예다함 상조서비스(티포트)  무형      0        NaN  2020      1    1   
38308        더케이 예다함 상조서비스(티포트)  무형      0        NaN  2020      1    1   

           time  
0      06:00:00  
1      06:00:00  
2      06:20:00  
3      06:20:00  
4      06:40:00  
...         ...  
38304  00:20:00  
38305  00:40:00  
38306  01:00:00  
38307  01:20:00  
38308  01:40:00  

[38309 rows x 12 columns]

## 1. 상품군 분석

In [8]:
category = pf['상품군'].unique()
len(category)

12

In [9]:
pf[pf['상품군'] == '무형']['판매단가'].unique()

array([0], dtype=int64)

In [10]:
pf[['상품코드', '상품군']].drop_duplicates()['상품군'].value_counts()

의류      362
잡화      309
가전      259
주방      245
속옷      238
생활용품    183
농수축     152
가구      114
무형       82
이미용      64
침구       60
건강기능     56
Name: 상품군, dtype: int64

### conclusion 
- 12개의 상품군 존재.
- 무형의 상품군은 모두 판매단가가 0

## 2. 마더코드 분석

In [11]:
mother_code = pf['마더코드'].unique()
len(mother_code)

716

In [12]:
sorted_mothercode = sorted(mother_code)

In [13]:
pf[pf['마더코드'] == sorted_mothercode[308]]

방송일시  노출(분)    마더코드    상품코드  \
13752 2019-05-11 09:00:00   20.0  100352  201141   
13753 2019-05-11 09:20:00   20.0  100352  201141   
13754 2019-05-11 09:40:00   20.0  100352  201141   
14390 2019-05-15 23:00:00   20.0  100352  201141   
14391 2019-05-15 23:20:00   20.0  100352  201141   
...                   ...    ...     ...     ...   
30083 2019-10-10 15:20:00   20.0  100352  201142   
30084 2019-10-10 15:40:00   20.0  100352  201142   
37689 2019-12-25 07:20:00   20.0  100352  201126   
37690 2019-12-25 07:40:00   20.0  100352  201126   
37691 2019-12-25 08:00:00   20.0  100352  201126   

                                  상품명 상품군     판매단가         취급액  year  month  \
13752         마르엘라로사티 휘메일 풀스킨 밍크 후드코트  의류  1590000  26382000.0  2019      5   
13753         마르엘라로사티 휘메일 풀스킨 밍크 후드코트  의류  1590000  68755000.0  2019      5   
13754         마르엘라로사티 휘메일 풀스킨 밍크 후드코트  의류  1590000  96820000.0  2019      5   
14390         마르엘라로사티 휘메일 풀스킨 밍크 후드코트  의류  1590000  40891000.0  2019      5   
14391         마르엘라로사티 휘메일 풀스킨 밍크 후드코트  의류  1590000  65060000.0  2019      5   
...                               ...  ..      ...         ...   ...    ...   
30083    마르엘라로사티 휘메일 풀스킨 밍크 후드코트(앱할인)  의류  1490000  25119000.0  2019     10   
30084    마르엘라로사티 휘메일 풀스킨 밍크 후드코트(앱할인)  의류  1490000  36973000.0  2019     10   
37689  마르엘라로사티 나파 휘메일 풀스킨 밍크하프코트(11월)  의류  1490000  17070000.0  2019     12   
37690  마르엘라로사티 나파 휘메일 풀스킨 밍크하프코트(11월)  의류  1490000  11264000.0  2019     12   
37691  마르엘라로사티 나파 휘메일 풀스킨 밍크하프코트(11월)  의류  1490000  46372000.0  2019     12   

       day      time  
13752   11  09:00:00  
13753   11  09:20:00  
13754   11  09:40:00  
14390   15  23:00:00  
14391   15  23:20:00  
...    ...       ...  
30083   10  15:20:00  
30084   10  15:40:00  
37689   25  07:20:00  
37690   25  07:40:00  
37691   25  08:00:00  

[160 rows x 12 columns]

In [14]:
pf[pf['상품군']=='이미용']['마더코드'].unique()

array([100074, 100774, 100212, 100267, 100266, 100446, 100529, 100211,
       100265, 100362, 100213, 100680, 100775, 100681, 100399, 100628,
       100773, 100319, 100766, 100244, 100000, 100332, 100327, 100045,
       100711, 100258, 100260, 100672, 100245, 100314, 100247, 100685,
       100178, 100723, 100710, 100118, 100824, 100262, 100438, 100451,
       100697, 100541], dtype=int64)

### conclusion
- 716개의 마더코드 존재.
- 마더코드는 대체로 브랜드 기준으로 나뉨. 하지만 같은 브랜드가 다른 마더코드를 가지기도 함.

In [15]:
goods_code = pf['상품코드'].unique()
len(goods_code)

2124

In [16]:
code_price = pf[['상품명', '판매단가']].drop_duplicates()
len(code_price)

1959

### 요일 추가
1-일요일, 2-월요일,..., 7-토요일

In [17]:
pf['weekday'] = pf['방송일시'].dt.dayofweek
pf.head()

방송일시  노출(분)    마더코드    상품코드            상품명 상품군   판매단가  \
0 2019-01-01 06:00:00   20.0  100346  201072  테이트 남성 셀린니트3종  의류  39900   
1 2019-01-01 06:00:00    NaN  100346  201079  테이트 여성 셀린니트3종  의류  39900   
2 2019-01-01 06:20:00   20.0  100346  201072  테이트 남성 셀린니트3종  의류  39900   
3 2019-01-01 06:20:00    NaN  100346  201079  테이트 여성 셀린니트3종  의류  39900   
4 2019-01-01 06:40:00   20.0  100346  201072  테이트 남성 셀린니트3종  의류  39900   

         취급액  year  month  day      time  weekday  
0  2099000.0  2019      1    1  06:00:00        1  
1  4371000.0  2019      1    1  06:00:00        1  
2  3262000.0  2019      1    1  06:20:00        1  
3  6955000.0  2019      1    1  06:20:00        1  
4  6672000.0  2019      1    1  06:40:00        1

In [22]:
pf.loc[pf['노출(분)'].isnull(), '노출(분)'] = \
              pf.loc[pf['노출(분)'].isnull(), '방송일시'].map(pf.loc[pf['노출(분)'].notnull()]\
                                                      .set_index('방송일시')['노출(분)'])

In [29]:
pf[60:70]

방송일시  노출(분)    마더코드    상품코드                  상품명   상품군  \
60 2019-01-02 00:20:00   20.0  100377  201225  그렉노먼 남성 캐시터치 터틀넥 4종    의류   
61 2019-01-02 00:20:00   20.0  100377  201231  그렉노먼 여성 캐시터치 터틀넥 4종    의류   
62 2019-01-02 00:40:00   20.0  100377  201225  그렉노먼 남성 캐시터치 터틀넥 4종    의류   
63 2019-01-02 00:40:00   20.0  100377  201231  그렉노먼 여성 캐시터치 터틀넥 4종    의류   
64 2019-01-02 01:00:00   20.0  100193  200652   레이프릴 무빙 맥시풀커버 브라팬티    속옷   
65 2019-01-02 01:20:00   20.0  100193  200652   레이프릴 무빙 맥시풀커버 브라팬티    속옷   
66 2019-01-02 01:40:00   17.1  100193  200652   레이프릴 무빙 맥시풀커버 브라팬티    속옷   
67 2019-01-02 06:00:00   20.0  100229  200800       벨라홈 논슬립 옷걸이 세트  생활용품   
68 2019-01-02 06:20:00   20.0  100229  200800       벨라홈 논슬립 옷걸이 세트  생활용품   
69 2019-01-02 06:40:00   20.0  100229  200800       벨라홈 논슬립 옷걸이 세트  생활용품   

     판매단가         취급액  year  month  day      time  weekday  
60  69900  11985000.0  2019      1    2  00:20:00        2  
61  69900   8491000.0  2019      1    2  00:20:00        2  
62  69900  16461000.0  2019      1    2  00:40:00        2  
63  69900  10641000.0  2019      1    2  00:40:00        2  
64  69900   9647000.0  2019      1    2  01:00:00        2  
65  69900  15516000.0  2019      1    2  01:20:00        2  
66  69900  18695000.0  2019      1    2  01:40:00        2  
67  29800   9605000.0  2019      1    2  06:00:00        2  
68  29800  11265000.0  2019      1    2  06:20:00        2  
69  29800  14371000.0  2019      1    2  06:40:00        2